newIDE/app/src/InstancesEditor/InstancesRenderer/LayerRenderer.js

```diff
@@ -134,22 +141,102 @@ export default class LayerRenderer {
     );
   };
 
-  getInstanceWidth = (instance: gdInitialInstance) => {
+  getUnrotatedInstanceWidth = (instance: gdInitialInstance) => {
     if (instance.hasCustomSize()) return instance.getCustomWidth();
 
     return this.renderedInstances[instance.ptr]
       ? this.renderedInstances[instance.ptr].getDefaultWidth()
       : 0;
   };
 
-  getInstanceHeight = (instance: gdInitialInstance) => {
+  getUnrotatedInstanceHeight = (instance: gdInitialInstance) => {
     if (instance.hasCustomSize()) return instance.getCustomHeight();
 
     return this.renderedInstances[instance.ptr]
       ? this.renderedInstances[instance.ptr].getDefaultHeight()
       : 0;
   };
 
+  getInstanceOBB(instance: gdInitialInstance, bounds: Rectangle): Rectangle {
+    const left = this.getUnrotatedInstanceLeft(instance);
+    const top = this.getUnrotatedInstanceTop(instance);
+    const right = left + this.getUnrotatedInstanceWidth(instance);
+    const bottom = top + this.getUnrotatedInstanceHeight(instance);
+
+    bounds.left = left;
+    bounds.right = right;
+    bounds.top = top;
+    bounds.bottom = bottom;
+    return bounds;
+  }
+
+  getRectanglePath(instance: gdInitialInstance): Array<[number, number]> {
+    const left = this.getUnrotatedInstanceLeft(instance);
+    const top = this.getUnrotatedInstanceTop(instance);
+    const right = left + this.getUnrotatedInstanceWidth(instance);
+    const bottom = top + this.getUnrotatedInstanceHeight(instance);
+
+    this.temporaryRectanglePath[0][0] = left;
+    this.temporaryRectanglePath[0][1] = top;
+
+    this.temporaryRectanglePath[1][0] = left;
+    this.temporaryRectanglePath[1][1] = bottom;
+
+    this.temporaryRectanglePath[2][0] = right;
+    this.temporaryRectanglePath[2][1] = bottom;
+
+    this.temporaryRectanglePath[3][0] = right;
+    this.temporaryRectanglePath[3][1] = top;
+
+    return this.temporaryRectanglePath;
+  }
+
+  rotate(
```

Comment:
Let's move this outside of the class as a pure function (avoid accidental reuse of it outside of this context)